### PR TITLE
Add ANY_VALUE agg function

### LINKED ISSUE #4092: Support ANY_VALUE aggregate function
Google BigQuery supports the [any_value](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#any_value) aggregate function which returns the first non-null value encountered. We have a similar function (`first`), but it returns the first value encountered. We would like to add support for the `ANY_VALUE` function as well.

### PR DESCRIPTION
**Pull request by mauropagano**
Fixes #4092 

I tried to follow all the directions in `CONTRIBUTING` but this is my first PR here, apologies in advance if something is off.
I used and re-adjusted the tests from `last()` because it seemed to have a larger coverage.

I have two items I'm not too sure about though:
- `IgnoreNulls()` inside `FirstState`, I'm not sure if it's useful to add a param to the template and return that. I did it for `FirstVectorFunction` but it seemed more natural there
- About the comparison with BigQuery, the behavior is a bit different when it comes to arrays because for BigQuery `NULL` is basically `[NULL]` but not for DuckDB. The current change considers only `NULL` and not `[NULL]` as something to skip. 

There is still the issue mentioned in #4092 about `UNION ALL` blocks but I guess that applies to `last()` so maybe it could be another issue / PR ?

### CODE REVIEW

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good, one minor comment:

**Comment by Mytherin** [comment]
Could you also have a look at the vector sizes test that is failing?

File: src/catalog/default/default_functions.cpp [hunk 1]
```diff
@@ -113,6 +113,7 @@ static DefaultMacro internal_macros[] = {
 	{DEFAULT_SCHEMA, "list_entropy", {"l", nullptr}, "list_aggr(l, 'entropy')"},
 	{DEFAULT_SCHEMA, "list_last", {"l", nullptr}, "list_aggr(l, 'last')"},
 	{DEFAULT_SCHEMA, "list_first", {"l", nullptr}, "list_aggr(l, 'first')"},
+	{DEFAULT_SCHEMA, "list_any_value", {"l", nullptr}, "list_aggr(l, 'any_value')"},
```
[the remark this answers is not in the document]
Comment (mauropagano) [reply src/catalog/default/default_functions.cpp:116]: Added! Sorry I missed it the first time around

**Comment by mauropagano** [comment]
(please let me know if Discord is a better place for these discussion)

About the vector size test failing, I can reproduce locally setting `STANDARD_VECTOR_SIZE` to 2 but I'm not totally sure I understand the failure (btw I think `skewness()` and `kurtosis()` fail too with this small vector size).

I get where the failure comes from and it does work mapping `"any_value"` in `transform_function.cpp` but I have a couple doubts:
- why does `any_value()` work fine with larger (default) `STANDARD_VECTOR_SIZE`. I'm assuming this is intended but I'm not sure I understand why. The type of window functions explicitly defined as `WINDOW_*` is a small set vs the regular aggregate functions, does it mean they all would have the same behavior? 
- on how to solve this, should `any_value` become a new `ExpressionType`, i.e. `WINDOW_ANY_VALUE`? Asking since `any_value()` in window function is basically `first(... ignore nulls)`, so not sure if I should try to somewhat overload `WINDOW_FIRST_VALUE`.

**Comment by Mytherin** [comment]
Window functions in general do not work with vector sizes of 2 currently because of the way the window segment tree is setup. That is unrelated to this PR and applies to all window function aggregates, as you have noticed as well. All you need to do is add `require vector_size 512` to the test.

The `WINDOW_` aggregates are special aggregates that only exist for window functions - all regular aggregates automatically work with window functions as well. Nothing is required to get that to work aside from implementing our aggregate function interface.

**Comment by mauropagano** [comment]
Thanks! 

I'm not sure about the two tests that still fail, I think "CodeQuality" used to pass and from the log I'm not sure if it was transitory or not, I can't seem to be able to restart it myself

**Comment by Mytherin** [comment]
Looks like they were spurious failures. Thanks for the PR!

**Pull request merged**